test/test_binary_ufuncs.py

```diff
@@ -1359,100 +1359,109 @@ def test_rdiv(self, device, dtype):
-                self.assertEqual(x_ == -1, x < 0)
+    def test_fmod_remainder_by_zero_integral(self, device, dtype):
+        fn_list = [(torch.fmod, lambda x, mod: x.fmod_(mod)),
+                   (torch.remainder, lambda x, mod: x.remainder_(mod))]
+        for fns in fn_list:
+            # check integral tensor fmod/remainder to zero
+            x = make_tensor((10, 10), device=device, dtype=dtype, low=-9, high=9)
+            zero = torch.zeros_like(x)
+            # out
+            out = torch.empty(0, device=device, dtype=dtype)
+            # In-place
+            x_ = x.clone()
+            # RuntimeError on CPU
+            if device == 'cpu':
+                with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
+                    fns[0](x, zero)
+                with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
+                    fns[0](x, zero, out=out)
+                with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
+                    fns[1](x_, zero)
+            # Different value for different dtype on GPU
             else:
-                value = 255 if dtype == torch.uint8 else -1
-                self.assertTrue(torch.all(x.fmod(zero) == value))
-                # out
-                torch.fmod(x, zero, out=out)
-                self.assertTrue(torch.all(out == value))
-                self.assertEqual(out.size(), torch.Size([10, 10]))
-                # in-place
-                x_.fmod_(zero)
-                self.assertTrue(torch.all(x_ == value))
+                if dtype == torch.int64:
+                    self.assertEqual(fns[0](x, zero) == 4294967295, x >= 0)
+                    self.assertEqual(fns[0](x, zero) == -1, x < 0)
+                    # out
+                    fns[0](x, zero, out=out)
+                    self.assertEqual(out == 4294967295, x >= 0)
+                    self.assertEqual(out == -1, x < 0)
+                    self.assertEqual(out.size(), torch.Size([10, 10]))
+                    # in-place
+                    fns[1](x_, zero)
+                    self.assertEqual(x_ == 4294967295, x >= 0)
+                    self.assertEqual(x_ == -1, x < 0)
+                else:
+                    value = 255 if dtype == torch.uint8 else -1
+                    self.assertTrue(torch.all(fns[0](x, zero) == value))
+                    # out
+                    fns[0](x, zero, out=out)
+                    self.assertTrue(torch.all(out == value))
+                    self.assertEqual(out.size(), torch.Size([10, 10]))
+                    # in-place
+                    fns[1](x_, zero)
+                    self.assertTrue(torch.all(x_ == value))
 
     @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False, include_bool=False, include_complex=False))
-    def test_fmod(self, device, dtype):
+    def test_fmod_remainder(self, device, dtype):
         # Use numpy as reference
         def _reference_implementation(x, mod):
```

Comment:
Once again, "reference_implementation" seems like a misnomer because this function actually performs the test.